graphene_sqlalchemy/filters.py

```diff
@@ -0,0 +1,180 @@
+    filtering methods ("eq, nEq") for different types of scalars.
+    The Dynamic fields will resolve to Meta.filtered_type"""
+
+    @classmethod
+    def __init_subclass_with_meta__(cls, type=None, _meta=None, **options):
+
+        # get all filter functions
+        filter_function_regex = re.compile(".+_filter$")
+
+        filter_functions = []
+
+        # Search the entire class for functions matching the filter regex
+        for func in dir(cls):
+            func_attr = getattr(cls, func)
+            # Check if attribute is a function
+            if callable(func_attr) and filter_function_regex.match(func):
+                # add function and attribute name to the list
+                filter_functions.append((
+                    re.sub("\_filter$", "", func), func_attr.__annotations__)
+                )
+
+        # Init meta options class if it doesn't exist already
+        if not _meta:
+            _meta = InputObjectTypeOptions(cls)
+
+        new_filter_fields = {}
+
+        # Generate Graphene Fields from the filter functions based on type hints
+        for field_name, annotations in filter_functions:
+            assert "val" in annotations, "Each filter method must have a value field with valid type annotations"
+            # If type is generic, replace with actual type of filter class
+            if annotations["val"] == AbstractType:
+                # TODO Maybe there is an existing class or a more elegant way to solve this
+                # One option would be to only annotate non-abstract filters
+                new_filter_fields.update({field_name: graphene.InputField(type)})
+            else:
+                # TODO this is a place holder, we need to convert the type of val to a valid graphene
+                # type that we can pass to the InputField. We could re-use converter.convert_hybrid_property_return_type
+                new_filter_fields.update({field_name: graphene.InputField(graphene.String)})
+
+        # Add all fields to the meta options. graphene.InputbjectType will take care of the rest
+        if _meta.fields:
+            _meta.fields.update(new_filter_fields)
+        else:
+            _meta.fields = new_filter_fields
+
+        # Pass modified meta to the super class
+        super(FieldFilter, cls).__init_subclass_with_meta__(_meta=_meta, **options)
+
+    # Abstract methods can be marked using AbstractType. See comment on the init method
+    @classmethod
+    def eq_filter(cls, val: AbstractType) -> bool:
+        # TBD filtering magic
+        pass
+
+
+class StringFilter(FieldFilter):
+    class Meta:
+        type = graphene.String
+
+
+class BooleanFilter(FieldFilter):
+    class Meta:
+        type = graphene.Boolean
+
+
+class NumberFilter(FieldFilter):
+    """Intermediate Filter class since all Numbers are in an order relationship (support <, > etc)"""
+
+    class Meta:
+        abstract = True
+
+    @classmethod
+    def gt_filter(cls, val: AbstractType) -> bool:
```

Comment:
Should this live on `NumberFilter`? Technically strings are ordered as well. Maybe we can call this `OrderedFilter` and numbers, strings, etc. can inherit?

Reply:
Great point, added!